qiskit_nature/second_q/algorithms/excited_states_solvers/qeom.py

```diff
@@ -145,97 +240,77 @@ def solve(
         Args:
             problem: A class encoding a problem to be solved.
             aux_operators: Additional auxiliary operators to evaluate.
-
         Returns:
             An interpreted :class:`~.EigenstateResult`. For more information see also
             :meth:`~.BaseProblem.interpret`.
         """
 
-        if aux_operators is not None:
-            logger.warning(
-                "With qEOM the auxiliary operators can currently only be "
-                "evaluated on the ground state."
-            )
-
-        # 1. Run ground state calculation
-        groundstate_result = self._gsc.solve(problem, aux_operators)
-
-        # 2. Prepare the excitation operators
-        main_second_q_op, _ = problem.second_q_ops()
-
-        num_particles = getattr(problem, "num_particles", None)
-
-        self._untapered_qubit_op_main = self._gsc.qubit_converter.convert_only(
-            main_second_q_op, num_particles
+        # 1. Prepare all operators
+        (
+            pre_tap_qubit_op_main,  # Hamiltonian
+            pre_tap_default_aux_ops,  # Default auxiliary observables
+            pre_tap_custom_aux_ops,  # User-defined auxiliary observables
+        ) = self.get_qeom_qubit_operators(problem, aux_operators)
+
+        # 2. Run ground state calculation with fully tapered custom auxiliary operators
+        # Note that the solve() method includes the `second_q' auxiliary operators
+        tap_custom_aux_ops = self.qubit_converter._symmetry_reduce_clifford(
+            ListOrDict(pre_tap_custom_aux_ops), True
```

Comment:
Ideally you can isolate the usage of `ListOrDict` to the `QubitConverter` and avoid the need to wrap it here.